pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -416,10 +416,40 @@ func foreachServer(client *gophercloud.ServiceClient, opts servers.ListOptsBuild
 }
 
 func getServerByName(client *gophercloud.ServiceClient, name types.NodeName) (*servers.Server, error) {
+	glog.V(5).Infof("Looking up server by instance name : %v", name)
+	srv, err := getServerByInstanceName(client, name)
+	if err == nil {
+		glog.V(5).Infof("Found server : %v/%v", srv.ID, srv.Name)
+		return srv, err
+	}
+	glog.V(5).Infof("Looking up server by host name : %v", name)
+	return getServerByHostName(client, name)
+}
+
+type listOpts struct {
+	Hostname string `q:"hostname"`
```

Comment:
I'm not sure if searching for a hostname works like this? I don't see `hostname` in https://godoc.org/github.com/rackspace/gophercloud/openstack/compute/v2/servers#ListOpts. The nova API says
> This parameter is only valid when specified by administrators. If non-admin users specify this parameter, it is ignored.

See https://developer.openstack.org/api-ref/compute/#list-server-request

So I don't think we can use this filter. It shouldn't be necessary or even recommended to provide an admin user for installing k8s on OpenStack